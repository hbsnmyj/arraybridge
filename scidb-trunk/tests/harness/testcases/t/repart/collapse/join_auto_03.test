
--setup

create temp array heatmap<celsius:double>[h=0:9,5,0,m=0:9,5,0]
--igdata "store(build(heatmap,h*1.1*10+m*3.7), heatmap)"

create temp array square<v:int64>[s=0:9,5,0,q=0:9,5,0]
--igdata "store(build(square, s*100 + q), square)"

# This setup is from the Ref Guide's cross_between() example:
--start-query-logging
show(square)
show(heatmap)
--echo == J ops
--error --code scidb::SCIDB_SE_OPERATOR::SCIDB_LE_ALL_INPUTS_AUTOCHUNKED "join(repart(square, <v:int64>[s,q]), repart(heatmap, <celsius:double>[h,m]))"

join(square , heatmap)
join(repart(square, <v:int64>[s,q]), heatmap)
join(square, repart(heatmap, <celsius:double>[h,m]))


--stop-query-logging

--cleanup
--echo Cleanup...
remove(heatmap)
remove(square)
