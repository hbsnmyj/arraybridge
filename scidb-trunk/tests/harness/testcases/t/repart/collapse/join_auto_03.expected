Query was executed successfully

[Query was executed successfully, ignoring data output by this query.]

Query was executed successfully

[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <show(square)>
{i} schema
{0} 'square<v:int64> [s=0:9,5,0,q=0:9,5,0]'

SCIDB QUERY : <show(heatmap)>
{i} schema
{0} 'heatmap<celsius:double> [h=0:9,5,0,m=0:9,5,0]'

== J ops
SCIDB QUERY : <join(repart(square, <v:int64>[s,q]), repart(heatmap, <celsius:double>[h,m]))>
[An error expected at this place for the query "join(repart(square, <v:int64>[s,q]), repart(heatmap, <celsius:double>[h,m]))". And it failed with error code = scidb::SCIDB_SE_OPERATOR::SCIDB_LE_ALL_INPUTS_AUTOCHUNKED. Expected error code = scidb::SCIDB_SE_OPERATOR::SCIDB_LE_ALL_INPUTS_AUTOCHUNKED.]

SCIDB QUERY : <join(square , heatmap)>
{s,q} v,celsius
{0,0} 0,0
{0,1} 1,3.7
{0,2} 2,7.4
{0,3} 3,11.1
{0,4} 4,14.8
{1,0} 100,11
{1,1} 101,14.7
{1,2} 102,18.4
{1,3} 103,22.1
{1,4} 104,25.8
{2,0} 200,22
{2,1} 201,25.7
{2,2} 202,29.4
{2,3} 203,33.1
{2,4} 204,36.8
{3,0} 300,33
{3,1} 301,36.7
{3,2} 302,40.4
{3,3} 303,44.1
{3,4} 304,47.8
{4,0} 400,44
{4,1} 401,47.7
{4,2} 402,51.4
{4,3} 403,55.1
{4,4} 404,58.8
{0,5} 5,18.5
{0,6} 6,22.2
{0,7} 7,25.9
{0,8} 8,29.6
{0,9} 9,33.3
{1,5} 105,29.5
{1,6} 106,33.2
{1,7} 107,36.9
{1,8} 108,40.6
{1,9} 109,44.3
{2,5} 205,40.5
{2,6} 206,44.2
{2,7} 207,47.9
{2,8} 208,51.6
{2,9} 209,55.3
{3,5} 305,51.5
{3,6} 306,55.2
{3,7} 307,58.9
{3,8} 308,62.6
{3,9} 309,66.3
{4,5} 405,62.5
{4,6} 406,66.2
{4,7} 407,69.9
{4,8} 408,73.6
{4,9} 409,77.3
{5,0} 500,55
{5,1} 501,58.7
{5,2} 502,62.4
{5,3} 503,66.1
{5,4} 504,69.8
{6,0} 600,66
{6,1} 601,69.7
{6,2} 602,73.4
{6,3} 603,77.1
{6,4} 604,80.8
{7,0} 700,77
{7,1} 701,80.7
{7,2} 702,84.4
{7,3} 703,88.1
{7,4} 704,91.8
{8,0} 800,88
{8,1} 801,91.7
{8,2} 802,95.4
{8,3} 803,99.1
{8,4} 804,102.8
{9,0} 900,99
{9,1} 901,102.7
{9,2} 902,106.4
{9,3} 903,110.1
{9,4} 904,113.8
{5,5} 505,73.5
{5,6} 506,77.2
{5,7} 507,80.9
{5,8} 508,84.6
{5,9} 509,88.3
{6,5} 605,84.5
{6,6} 606,88.2
{6,7} 607,91.9
{6,8} 608,95.6
{6,9} 609,99.3
{7,5} 705,95.5
{7,6} 706,99.2
{7,7} 707,102.9
{7,8} 708,106.6
{7,9} 709,110.3
{8,5} 805,106.5
{8,6} 806,110.2
{8,7} 807,113.9
{8,8} 808,117.6
{8,9} 809,121.3
{9,5} 905,117.5
{9,6} 906,121.2
{9,7} 907,124.9
{9,8} 908,128.6
{9,9} 909,132.3

SCIDB QUERY : <join(repart(square, <v:int64>[s,q]), heatmap)>
{s,q} v,celsius
{0,0} 0,0
{0,1} 1,3.7
{0,2} 2,7.4
{0,3} 3,11.1
{0,4} 4,14.8
{1,0} 100,11
{1,1} 101,14.7
{1,2} 102,18.4
{1,3} 103,22.1
{1,4} 104,25.8
{2,0} 200,22
{2,1} 201,25.7
{2,2} 202,29.4
{2,3} 203,33.1
{2,4} 204,36.8
{3,0} 300,33
{3,1} 301,36.7
{3,2} 302,40.4
{3,3} 303,44.1
{3,4} 304,47.8
{4,0} 400,44
{4,1} 401,47.7
{4,2} 402,51.4
{4,3} 403,55.1
{4,4} 404,58.8
{0,5} 5,18.5
{0,6} 6,22.2
{0,7} 7,25.9
{0,8} 8,29.6
{0,9} 9,33.3
{1,5} 105,29.5
{1,6} 106,33.2
{1,7} 107,36.9
{1,8} 108,40.6
{1,9} 109,44.3
{2,5} 205,40.5
{2,6} 206,44.2
{2,7} 207,47.9
{2,8} 208,51.6
{2,9} 209,55.3
{3,5} 305,51.5
{3,6} 306,55.2
{3,7} 307,58.9
{3,8} 308,62.6
{3,9} 309,66.3
{4,5} 405,62.5
{4,6} 406,66.2
{4,7} 407,69.9
{4,8} 408,73.6
{4,9} 409,77.3
{5,0} 500,55
{5,1} 501,58.7
{5,2} 502,62.4
{5,3} 503,66.1
{5,4} 504,69.8
{6,0} 600,66
{6,1} 601,69.7
{6,2} 602,73.4
{6,3} 603,77.1
{6,4} 604,80.8
{7,0} 700,77
{7,1} 701,80.7
{7,2} 702,84.4
{7,3} 703,88.1
{7,4} 704,91.8
{8,0} 800,88
{8,1} 801,91.7
{8,2} 802,95.4
{8,3} 803,99.1
{8,4} 804,102.8
{9,0} 900,99
{9,1} 901,102.7
{9,2} 902,106.4
{9,3} 903,110.1
{9,4} 904,113.8
{5,5} 505,73.5
{5,6} 506,77.2
{5,7} 507,80.9
{5,8} 508,84.6
{5,9} 509,88.3
{6,5} 605,84.5
{6,6} 606,88.2
{6,7} 607,91.9
{6,8} 608,95.6
{6,9} 609,99.3
{7,5} 705,95.5
{7,6} 706,99.2
{7,7} 707,102.9
{7,8} 708,106.6
{7,9} 709,110.3
{8,5} 805,106.5
{8,6} 806,110.2
{8,7} 807,113.9
{8,8} 808,117.6
{8,9} 809,121.3
{9,5} 905,117.5
{9,6} 906,121.2
{9,7} 907,124.9
{9,8} 908,128.6
{9,9} 909,132.3

SCIDB QUERY : <join(square, repart(heatmap, <celsius:double>[h,m]))>
{s,q} v,celsius
{0,0} 0,0
{0,1} 1,3.7
{0,2} 2,7.4
{0,3} 3,11.1
{0,4} 4,14.8
{1,0} 100,11
{1,1} 101,14.7
{1,2} 102,18.4
{1,3} 103,22.1
{1,4} 104,25.8
{2,0} 200,22
{2,1} 201,25.7
{2,2} 202,29.4
{2,3} 203,33.1
{2,4} 204,36.8
{3,0} 300,33
{3,1} 301,36.7
{3,2} 302,40.4
{3,3} 303,44.1
{3,4} 304,47.8
{4,0} 400,44
{4,1} 401,47.7
{4,2} 402,51.4
{4,3} 403,55.1
{4,4} 404,58.8
{0,5} 5,18.5
{0,6} 6,22.2
{0,7} 7,25.9
{0,8} 8,29.6
{0,9} 9,33.3
{1,5} 105,29.5
{1,6} 106,33.2
{1,7} 107,36.9
{1,8} 108,40.6
{1,9} 109,44.3
{2,5} 205,40.5
{2,6} 206,44.2
{2,7} 207,47.9
{2,8} 208,51.6
{2,9} 209,55.3
{3,5} 305,51.5
{3,6} 306,55.2
{3,7} 307,58.9
{3,8} 308,62.6
{3,9} 309,66.3
{4,5} 405,62.5
{4,6} 406,66.2
{4,7} 407,69.9
{4,8} 408,73.6
{4,9} 409,77.3
{5,0} 500,55
{5,1} 501,58.7
{5,2} 502,62.4
{5,3} 503,66.1
{5,4} 504,69.8
{6,0} 600,66
{6,1} 601,69.7
{6,2} 602,73.4
{6,3} 603,77.1
{6,4} 604,80.8
{7,0} 700,77
{7,1} 701,80.7
{7,2} 702,84.4
{7,3} 703,88.1
{7,4} 704,91.8
{8,0} 800,88
{8,1} 801,91.7
{8,2} 802,95.4
{8,3} 803,99.1
{8,4} 804,102.8
{9,0} 900,99
{9,1} 901,102.7
{9,2} 902,106.4
{9,3} 903,110.1
{9,4} 904,113.8
{5,5} 505,73.5
{5,6} 506,77.2
{5,7} 507,80.9
{5,8} 508,84.6
{5,9} 509,88.3
{6,5} 605,84.5
{6,6} 606,88.2
{6,7} 607,91.9
{6,8} 608,95.6
{6,9} 609,99.3
{7,5} 705,95.5
{7,6} 706,99.2
{7,7} 707,102.9
{7,8} 708,106.6
{7,9} 709,110.3
{8,5} 805,106.5
{8,6} 806,110.2
{8,7} 807,113.9
{8,8} 808,117.6
{8,9} 809,121.3
{9,5} 905,117.5
{9,6} 906,121.2
{9,7} 907,124.9
{9,8} 908,128.6
{9,9} 909,132.3

Cleanup...
Query was executed successfully

Query was executed successfully

